Update formautofill.properties from mozilla-central@e1f660fc493b

scottwu opened the pull request:
Updating the strings from formautofill for translation.

scottwu committed:
Update formautofill.properties from mozilla-central@e1f660fc493b

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
 preferenceGroupTitle = Form Autofill
-enableProfileAutofill = Enable Profile Autofill
-savedProfiles = Saved Profiles…
+enableAddressAutofill = Autofill addresses
```
This is OK until we enable the project in Pontoon. From that moment, the [same rules](https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_content_best_practices#Changing_existing_strings) as mozilla-central apply: string changes like this require new IDs.

As a general note through out the entire file: you need to add localization comments to explain the context, and use more intuitive string IDs. What is this? Where it's used? I have no idea if it's a title, a checkbox, something else.

Localizers see one string at a time and have no visibility in code or previous discussions.

scottwu replied:
I'm going through the IDs one by one and try to make their IDs more intuitive or add comments. And yes we will make sure we change the ID if there's any string change in the future.

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
-savedProfiles = Saved Profiles…
+enableAddressAutofill = Autofill addresses
+learnMore = Learn more
+savedAddresses = Saved Addresses…
```
What's the context here? Without any help, it either looks like a message to say that you saved addresses, and no ellipsis are needed, or "Saving Addresses…".

If it's a button label to access saved addresses, you should use a more intuitive ID (e.g. savedAddressesBtnLabel). That would also prevent developers from reusing the same string in a completely different context.

scottwu replied:
`savedAddressesBtnLabel` sounds good!

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
 updateAddressMessage = Would you like to update your address with this new information?
 createAddressLabel = Create New Address
 updateAddressLabel = Update Address
+saveCreditCardMessage = Would you like Firefox to save this credit card? (Security code will not be saved)
```
Avoid hard coding Firefox.

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
+enableAddressAutofill = Autofill addresses
+learnMore = Learn more
+savedAddresses = Saved Addresses…
+enableCreditCardAutofill = Autofill credit cards
```
Same here, context. What's this? A checkbox in preferences, a title, a button? Please add a localization comment.

scottwu replied:
I think I'll make it `autofillCreditCardsCheckbox`.

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
+learnMore = Learn more
+savedAddresses = Saved Addresses…
+enableCreditCardAutofill = Autofill credit cards
+savedCreditCards = Saved Credit Cards…
```
As above, if it's a button label, use a clearer ID and/or add a localization comment.

scottwu replied:
Will make it `savedCreditCardsBtnLabel` instead.

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
+savedAddresses = Saved Addresses…
+enableCreditCardAutofill = Autofill credit cards
+savedCreditCards = Saved Credit Cards…
 saveAddressesMessage = Firefox now saves addresses so you can fill out forms faster.
```
Don't hard code Firefox in strings.

scottwu replied:
Got it.

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
 saveAddressesMessage = Firefox now saves addresses so you can fill out forms faster.
-viewAutofillOptionsLink = View Form Autofill Options
+autofillOptionsLink = Form Autofill Options
+autofillSecurityOptionsLink = Form Autofill & Security Options
```
Are these two links? Where are they displayed?

scottwu replied:
They are used as links on the doorhangers, one is for [address](https://mozilla.invisionapp.com/share/AP8TFZ22G#/screens/190747959) and the other for [credit card](https://mozilla.invisionapp.com/share/7ZA4WEK9W#/screens/215537981). Do you think it's a good idea to prefix where the string will be used on all the IDs? (ex. `doorhanger.`, `pref.`, `dropdown.`) Would there be cases where being too specific is bad?

flodolo replied:
No need, explaining how they are used in a localization note is enough.

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -3,26 +3,36 @@
-viewAutofillOptionsLink = View Form Autofill Options
+autofillOptionsLink = Form Autofill Options
+autofillSecurityOptionsLink = Form Autofill & Security Options
 changeAutofillOptions = Change Form Autofill Options
```
Is this a title? Again, a better string ID would help.

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -31,3 +41,40 @@ fieldNameSeparator = ,\u0020
 # The text would be e.g. Also fill company, phone, email
```
The example doesn't match the actual string. Also, in which context is phishingWarningMessage used instead of phishingWarningMessage2?

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -31,3 +41,40 @@ fieldNameSeparator = ,\u0020
+nameOnCard = Name on Card
+cardExpires = Expires
+
+insecureFieldWarningDescription = %S has detected an insecure site. Credit card autofill is temporarily disabled
```
Explain what %S is in a localization comment

flodolo reviewed (changes requested):
A more general question: is Autofill supposed to be kept in English or translated? Is it "Form Autofill" or "Autofill"? You are not using them consistently.

scottwu commented:
Thanks again @flodolo, this is very helpful. I'll update the patch with renamed IDs and more localization notes.

scottwu commented on locales/en-US/formautofill.properties:
```diff
@@ -2,32 +2,116 @@
+# suggestion when users try to autofill credit card on an insecure website (without https).
+insecureFieldWarningDescription = %S has detected an insecure site. Form Autofill is temporarily disabled
+
+# LOCALIZATION NOTE: These strings are used in browser preferences.
```
Could localization note be used this way without attaching it to a particular string ID?

flodolo replied:
Tools are still relatively stupid, so they associate the comment to the following string. Ideally, you should specify the ID in the comment. See https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_content_best_practices#Add_localization_notes

scottwu commented:
I've changed some string ID to make them more self-explanatory, and added localization notes to most of the strings. Please let me know if this is would be clear enough for localizers. Thanks!

flodolo commented on locales/en-US/formautofill.properties:
```diff
@@ -2,32 +2,118 @@
+email = Email
+cancelBtnLabel = Cancel
+saveBtnLabel = Save
+countryWarningMessage = Form Autofill is currently available only for US addresses
```
Is this still true if we're enabling localization?

scottwu replied:
Yes, but we are **[planning](https://docs.google.com/document/d/1cmS1IKxXCd6tvOS8nRD6NwaoB_IWL2zMTl5nF0pOLts/edit)** to support more countries. In addition to having localization ready, we want to make sure our heuristic algorithm works well in the countries we are supporting.

flodolo reviewed (commented):
It looks definitely good, still pending two questions.

1. Clarifications on deadlines, how often localization is updated, etc.
2. Clarification on "Form Autofill": can it be translated? I definitely hope so

scottwu commented:
We are currently riding the train, even though can ship with Go Faster if we find compelling reason to do so in the future. So I guess the update frequency is same as regular release cycle. And yes, "Form Autofill" could be translated. When `autofill` is used by itself then it's the action of filling data automatically.
